cli/cmd/describe.go

```diff
@@ -448,44 +393,134 @@ func printPodsInfo(pods *corev1.PodList, instrumentationInstances *odigosv1.Inst
 	}
 }
 
-// installCmd represents the install command
 var describeCmd = &cobra.Command{
 	Use:   "describe",
 	Short: "Show details of a specific odigos entity",
 	Long:  `Print detailed description of a specific odigos entity, which can be used to troubleshoot issues`,
+}
+
+var describeSourceCmd = &cobra.Command{
+	Use:   "source",
+	Short: "Show details of a specific odigos source",
+	Long:  `Print detailed description of a specific odigos source, which can be used to troubleshoot issues`,
+}
+
+func printDescribeSource(ctx context.Context, client *kube.Client, workloadObj *K8sSourceObject) {
+	namespace, instrumentationConfig, instrumentedApplication, instrumentationInstances, pods, err := getRelevantResources(ctx, client, workloadObj)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		return
+	}
+
+	instrumented := printWorkloadManifestInfo(workloadObj, namespace)
+	printInstrumentationConfigInfo(instrumentationConfig, instrumented)
+	printInstrumentedApplicationInfo(instrumentedApplication, instrumented)
+	containerNameToExpectedDevices := printAppliedInstrumentationDeviceInfo(workloadObj, instrumentedApplication, instrumented)
+	printPodsInfo(pods, instrumentationInstances, containerNameToExpectedDevices)
+}
+
+var describeSourceDeploymentCmd = &cobra.Command{
+	Use:     "deployment <name>",
+	Short:   "Show details of a specific odigos source of type deployment",
+	Long:    `Print detailed description of a specific odigos source of type deployment, which can be used to troubleshoot issues`,
+	Aliases: []string{"deploy", "deployments", "deploy.apps", "deployment.apps", "deployments.apps"},
+	Args:    cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 
 		client, err := kube.CreateClient(cmd)
 		if err != nil {
 			kube.PrintClientErrorAndExit(err)
 		}
+
 		ctx := cmd.Context()
+		name := args[0]
 		ns := cmd.Flag("namespace").Value.String()
-
-		if len(args) != 2 {
-			fmt.Println("Usage: odigos describe <kind> <name>")
+		deployment, err := client.AppsV1().Deployments(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
 			return
 		}
+		workloadObj := &K8sSourceObject{
+			Kind:            "deployment",
```

Comment:
Not sure but I think deployment.Kind returns this string.
Also other places

Reply:
Unfortunately, The `Kind` property is not populated by the client when instantiating a `Deployment` object. This is why I needed to add it explicitly